packages/form-validation/src/validation.js

```diff
@@ -144,6 +148,9 @@ export default class AuroFormValidation {
         this.validateType(elem);
         this.validateAttributes(elem);
       }
+    } else if (elem.value === undefined) {
```

Comment:
We are touching validity to reset it. In the case of this fix, users expect that validity state to reset when setting the element to `value=undefined`. Since `validationShouldRun` is false, it skips the validation logic and resets the validity state.

Reply:
should we not reset the validity always at the beginning of the validate function? as `elem.value` is not only case that we want to reset.